### PR TITLE
hmem: Support CUDA cuPointerGetAttribute

### PR DESCRIPTION
**Pull request by iziemba**
cuPointerGetAttribute is required for a CUDA MR cache implementation.

Signed-off-by: Ian Ziemba <ian.ziemba@hpe.com>

### CODE REVIEW

File: src/hmem_cuda.c [hunk 1]
```diff
@@ -82,6 +87,12 @@ const char *ofi_cudaGetErrorString(cudaError_t error)
 	return cuda_ops.cudaGetErrorString(error);
 }
 
+CUresult ofi_cuPointerGetAttribute(void *data, CUpointer_attribute attribute,
```
Comment (rajachan) [inline src/hmem_cuda.c:90]: Have you come across any situation where we would need this outside of providers that want to handle GPU buffers without requiring FI_MR_HMEM?

Comment (iziemba) [reply src/hmem_cuda.c:90]: Yes. The CUDA memory monitor uses this function to get a buffer ID for a GPU address.

Comment (rajachan) [reply src/hmem_cuda.c:90]: Ah yes, we've talked about that. Looking at that PR next.

**Comment by shefty** [comment]
Travis CI failure is unrelated to this PR, but is real.  Captured failure in issue #5988 and restarted CI.  Intel CI went berserk.

**Pull request merged**